pkgs/collection/lib/src/iterable_extensions.dart

```diff
@@ -914,6 +914,33 @@ extension IterableIterableExtension<T> on Iterable<Iterable<T>> {
       };
 }
 
+/// Extension on iterables of [MapEntry].
+///
+/// An [Iterable<MapEntry>] is obtained using [Map.entries], these extensions
+/// make it easy to work on a [Map] as a list of pairs.
+extension IterableMapEntryExtension<K, V> on Iterable<MapEntry<K, V>> {
+  /// Creates a new lazy [Iterable] with all elements whose [MapEntry.key]
+  /// satisfy the predicate [test].
```

Comment:
Long first line. I'd also use a noun-phrased name to match the predicate named function.

Consider:
```dart
  /// The elements whose [MapEntry.key] values satisfy [test].
  ///
  /// The resulting iterable is lazily computing its elements 
  /// based on the elements this iterable.
```
Or something.

Reply:
I actually took the documentation string from `.where` which says:
```dart
>   /// Creates a new lazy [Iterable] with all elements that satisfy the
>   /// predicate [test].
>   ///
>   /// ...
```

I don't really care, the behavior is implied by the name, and the implementation is absolutely trivial to follow :D

Reply:
Yeah, that's too long too. But OK,  keep it here, it's not *that* too long.

But consider changing "elements" to "entries", since all elements of this iterable are (map-)entries, not just elements. (I'd refer to the elements of an `Iterable<num>` as "the numbers" too.)